### PR TITLE
Addwatch Operation

### PR DESCRIPTION
**Pull request by inf-rno**
- Add persistent, recursive watch functionality introduced in 3.6.0
- Upgrade setWatches to setWatches2

### CODE REVIEW

File: structs.go [hunk 1]
```diff
@@ -623,6 +641,10 @@ func requestStructForOp(op int32) interface{} {
 		return &SetDataRequest{}
 	case opSetWatches:
 		return &setWatchesRequest{}
+	case opSetWatches2:
```
Comment (shivnagarajan) [inline structs.go:644]: A nicer name would be good here.

File: constants.go [hunk 2]
```diff
@@ -32,6 +32,8 @@ const (
 	opClose           = -11
 	opSetAuth         = 100
 	opSetWatches      = 101
+	opSetWatches2     = 105
```
Comment (shivnagarajan) [inline constants.go:35]: what's up with values 102-104?

**Review by shivnagarajan — COMMENTED** [review]
mostly LGTM, weird I dont have an option to approve this.

Do we have a link to the upstream PR to merge this there?

Also do we have a link to the implementation in Java for this?

File: constants.go (hunk 2 above)
Comment (inf-rno) [reply constants.go:35]: theres a bunch of ops that the client doesnt support yet in the zk protocol

File: structs.go (hunk 1 above)
Comment (inf-rno) [reply structs.go:644]: this is actually the name of the op in zk. blame java i guess?

**Review by inf-rno — COMMENTED** [review]
I will need to clean this up a little and make it backwards compatible before i send it upstream. This is the feature that impls it in zookeeper https://github.com/apache/zookeeper/pull/1106

**Pull request merged**